### PR TITLE
[tests-only] [full-ci] additional test for share of matching file and folder name

### PR DESCRIPTION
**Pull request by sakshamgurung**
## Description

API test for
- [x] a user receives group shares for same name file/folder (user is member of multiple group)
- [x] a group receives shares for same name file/folder from multiple non-member users
- [x] a user receives share for file/folder (user decline the share first, then creates same name file/folder and at last accepts the share.)
- [x] additional test with different case name
- [x] add similar test for toShares

## Related Issue

- Additional API test for issue https://github.com/owncloud/core/issues/35414

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [ ] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>webUIAddUsers-chrome-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/31484/128/1

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>apiSharePublicLink2-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/31485/77/1

**Comment by SwikritiT** [comment]
@sakshamgurung Add [tests-only] in the title and you can also add [full-ci] so that the build won't be cancelled if certain pipeline fails.

File: tests/acceptance/features/apiShareManagementToRoot/acceptShares.feature [hunk 1]
```diff
@@ -731,3 +731,145 @@ Feature: accept/decline shares coming from internal users
...
+
+
+  Scenario: a user creates creates same name file after receiving share
+    Given user "Alice" has uploaded file with content "from alice" to "/message.txt"
+    When user "Alice" shares file "/message.txt" with user "Carol" using the sharing API
+    And user "Carol" accepts share "/message.txt" offered by user "Alice" using the sharing API
+    Then user "Carol" should see the following elements
+      | /message.txt |
+    And the content of file "/message.txt" for user "Carol" should be "from alice"
+    When user "Carol" declines share "/message.txt" offered by user "Alice" using the sharing API
```
Comment (SwikritiT) [inline tests/acceptance/features/apiShareManagementToRoot/acceptShares.feature:845]: Is it necessary to decline the share? Can't user `Carol` create new file withe the same name while share is in accepted state and check if both files are present and contents are also correct?

File: tests/acceptance/features/apiShareManagementToRoot/acceptShares.feature [hunk 2]
```diff
@@ -731,3 +731,145 @@ Feature: accept/decline shares coming from internal users
...
+
+  Scenario: a user creates creates same name folder after receiving share
+    Given user "Alice" has created folder "/PaRent"
+    And user "Alice" has uploaded file with content "from alice" to "/PaRent/message.txt"
+    When user "Alice" shares folder "/PaRent" with user "Carol" using the sharing API
+    And user "Carol" accepts share "/PaRent" offered by user "Alice" using the sharing API
+    Then user "Carol" should see the following elements
+      | /PaRent/ |
+    And the content of file "/PaRent/message.txt" for user "Carol" should be "from alice"
+    When user "Carol" declines share "/PaRent" offered by user "Alice" using the sharing API
```
Comment (SwikritiT) [inline tests/acceptance/features/apiShareManagementToRoot/acceptShares.feature:865]: Similar for this one too

File: tests/acceptance/features/apiShareManagementToRoot/acceptShares.feature [hunk 3]
```diff
@@ -731,3 +731,145 @@ Feature: accept/decline shares coming from internal users
...
+      | /PARENT/        |
+      | /PARENT (2)/    |
+      | /PARENT (3)/    |
+      | /PaRent/        |
+      | /PARENT.txt     |
+      | /PARENT (2).txt |
+      | /parent.txt     |
+    And the content of file "/PARENT (2)/parent.txt" for user "Brian" should be "subfile, from alice to grp2"
+    And the content of file "/PARENT.txt" for user "Brian" should be "from alice to grp2"
+
```
Comment (jasson99) [inline tests/acceptance/features/apiShareManagementToRoot/acceptShares.feature:786]: Can we check the content of the file shared by carol as well?

File: tests/acceptance/features/apiShareManagementToRoot/acceptShares.feature [hunk 4]
```diff
@@ -731,3 +731,145 @@ Feature: accept/decline shares coming from internal users
...
+      | /PARENT (3)/    |
+      | /PaRent/        |
+      | /PARENT.txt     |
+      | /PARENT (2).txt |
+      | /parent.txt     |
+    And the content of file "/PARENT (2)/parent.txt" for user "Carol" should be "subfile, from alice to grp1"
+    And the content of file "/PARENT.txt" for user "Carol" should be "from alice to grp1"
+
+
+  Scenario: a user creates creates same name file after receiving share
```
Comment (jasson99) [inline tests/acceptance/features/apiShareManagementToRoot/acceptShares.feature:838]:
```suggestion
  Scenario: a user creates same name file after receiving share
```

File: tests/acceptance/features/apiShareManagementToRoot/acceptShares.feature [hunk 5]
```diff
@@ -731,3 +731,145 @@ Feature: accept/decline shares coming from internal users
...
+    Given user "Alice" has uploaded file with content "from alice" to "/message.txt"
+    When user "Alice" shares file "/message.txt" with user "Carol" using the sharing API
+    And user "Carol" accepts share "/message.txt" offered by user "Alice" using the sharing API
+    Then user "Carol" should see the following elements
+      | /message.txt |
+    And the content of file "/message.txt" for user "Carol" should be "from alice"
```
Comment (jasson99) [inline tests/acceptance/features/apiShareManagementToRoot/acceptShares.feature:844]: Please check if we already have a scenario for this, "when a user accepts a share then it should be listed and the content should be as expected". If yes, we can replace this by
```suggestion
    And user "Carol" has accepted share "/message.txt" offered by user "Alice" using the sharing API
```

File: tests/acceptance/features/apiShareManagementToRoot/acceptShares.feature [hunk 6]
```diff
@@ -731,3 +731,145 @@ Feature: accept/decline shares coming from internal users
...
+      | /message.txt |
+    And the content of file "/message.txt" for user "Carol" should be "carol file"
+    When user "Carol" accepts share "/message.txt" offered by user "Alice" using the sharing API
+    Then user "Carol" should see the following elements
+      | /message.txt |
+      | /message (2).txt |
+    And the content of file "/message (2).txt" for user "Carol" should be "from alice"
+
+
+  Scenario: a user creates creates same name folder after receiving share
```
Comment (jasson99) [inline tests/acceptance/features/apiShareManagementToRoot/acceptShares.feature:857]:
```suggestion
  Scenario: a user creates same name folder after receiving share
```

File: tests/acceptance/features/apiShareManagementToRoot/acceptShares.feature [hunk 7]
```diff
@@ -731,3 +731,145 @@ Feature: accept/decline shares coming from internal users
...
+    Given user "Alice" has created folder "/PaRent"
+    And user "Alice" has uploaded file with content "from alice" to "/PaRent/message.txt"
+    When user "Alice" shares folder "/PaRent" with user "Carol" using the sharing API
+    And user "Carol" accepts share "/PaRent" offered by user "Alice" using the sharing API
+    Then user "Carol" should see the following elements
+      | /PaRent/ |
+    And the content of file "/PaRent/message.txt" for user "Carol" should be "from alice"
```
Comment (jasson99) [inline tests/acceptance/features/apiShareManagementToRoot/acceptShares.feature:864]: same here

File: tests/acceptance/features/apiShareManagementToRoot/acceptShares.feature (hunk 1 above)
Comment (sakshamgurung) [reply tests/acceptance/features/apiShareManagementToRoot/acceptShares.feature:845]: It is intentionally done to mimic some behavior.

File: tests/acceptance/features/apiShareManagementToRoot/acceptShares.feature [hunk 8]
```diff
@@ -731,3 +731,145 @@ Feature: accept/decline shares coming from internal users
...
+    And the content of file "/message.txt" for user "Carol" should be "from alice"
+    When user "Carol" declines share "/message.txt" offered by user "Alice" using the sharing API
+    And user "Carol" uploads file with content "carol file" to "/message.txt" using the WebDAV API
+    Then user "Carol" should see the following elements
+      | /message.txt |
+    And the content of file "/message.txt" for user "Carol" should be "carol file"
+    When user "Carol" accepts share "/message.txt" offered by user "Alice" using the sharing API
+    Then user "Carol" should see the following elements
+      | /message.txt |
+      | /message (2).txt |
```
Comment (saw-jan) [inline tests/acceptance/features/apiShareManagementToRoot/acceptShares.feature:853]:
```suggestion
      | /message.txt     |
      | /message (2).txt |
```

**Review by swoichha — COMMENTED** [review]
looks good to me

File: tests/acceptance/features/apiShareManagementToRoot/acceptShares.feature [hunk 9]
```diff
@@ -731,3 +731,142 @@ Feature: accept/decline shares coming from internal users
...
+    Then user "Carol" should see the following elements
+      | /PARENT/        |
+      | /PARENT (2)/    |
+      | /PARENT (3)/    |
+      | /PaRent/        |
+      | /PARENT.txt     |
+      | /PARENT (2).txt |
+      | /parent.txt     |
+    And the content of file "/PARENT (2)/parent.txt" for user "Carol" should be "subfile, from alice to grp1"
+    And the content of file "/PARENT.txt" for user "Carol" should be "from alice to grp1"
```
Comment (SwikritiT) [inline tests/acceptance/features/apiShareManagementToRoot/acceptShares.feature:838]: We can also check content of file shared by `Brian` similar to above scenario.

File: tests/acceptance/features/apiShareManagementToRoot/acceptShares.feature [hunk 10]
```diff
@@ -731,3 +731,142 @@ Feature: accept/decline shares coming from internal users
...
+
+
+  Scenario: a user creates same name file after receiving share
```
Comment (kiranparajuli589) [inline tests/acceptance/features/apiShareManagementToRoot/acceptShares.feature:841]:
```suggestion
  Scenario: user receives a share and uploads a file with the identical name
```

File: tests/acceptance/features/apiShareManagementToRoot/acceptShares.feature [hunk 11]
```diff
@@ -731,3 +731,142 @@ Feature: accept/decline shares coming from internal users
     When user "Alice" restores version index "1" of file "/toShareFile.txt" using the WebDAV API
     Then the content of file "/toShareFile.txt" for user "Alice" should be "Test Content."
     And the content of file "/toShareFile.txt" for user "Brian" should be "Test Content."
+
+
+  Scenario: a user receives multiple group share for matching file and folder name
```
Comment (kiranparajuli589) [inline tests/acceptance/features/apiShareManagementToRoot/acceptShares.feature:736]:
```suggestion
  Scenario: a user receives multiple group shares for matching file and folder name
```

**Review by Talank — COMMENTED** [review]
LGTM

**Review by saw-jan — COMMENTED** [review]
LGTM

**Review by phil-davis — APPROVED** [review]

**Pull request merged**